apps/bfd-server/bfd-server-war/src/main/java/gov/cms/bfd/server/war/r4/providers/DiagnosisUtilV2.java

```diff
@@ -0,0 +1,470 @@
+            C4BBClaimOutpatientInstitutionalDiagnosisType.EXTERNAL_CAUSE.getDisplay());
+
+      case REASONFORVISIT:
+        return new Coding(
+            C4BBClaimOutpatientInstitutionalDiagnosisType.PATIENT_REASON.getSystem(),
+            C4BBClaimOutpatientInstitutionalDiagnosisType.PATIENT_REASON.toCode(),
+            C4BBClaimOutpatientInstitutionalDiagnosisType.PATIENT_REASON.getDisplay());
+
+      case ADMITTING:
+      case OTHER:
+      default:
+        return new Coding(
+            C4BBClaimOutpatientInstitutionalDiagnosisType.OTHER.getSystem(),
+            C4BBClaimOutpatientInstitutionalDiagnosisType.OTHER.toCode(),
+            C4BBClaimOutpatientInstitutionalDiagnosisType.OTHER.getDisplay());
+    }
+  }
+
+  /**
+   * Translates from {@link DiagnosisLabel} to {@link ExDiagnosistype}
+   *
+   * <p>*Note* : Pharmacy EOBs do not get diagnosis mapped at this point in time. This is here for
+   * completeness and consistency.
+   */
+  static Coding translateLabelPharmacy(DiagnosisLabel label) {
+    switch (label) {
+      case PRINCIPAL:
+        return new Coding(
+            ExDiagnosistype.PRINCIPAL.getSystem(),
+            ExDiagnosistype.PRINCIPAL.toCode(),
+            ExDiagnosistype.PRINCIPAL.getDisplay());
+
+      case ADMITTING:
+        return new Coding(
+            ExDiagnosistype.ADMITTING.getSystem(),
+            ExDiagnosistype.ADMITTING.toCode(),
+            ExDiagnosistype.ADMITTING.getDisplay());
+
+      case FIRSTEXTERNAL:
+      case EXTERNAL:
+      case REASONFORVISIT:
+      case OTHER:
+      default:
+        return new Coding(label.getSystem(), label.toCode(), label.getDisplay());
+    }
+  }
+
+  /**
+   * Translates from {@link DiagnosisLabel} to {@link
+   * C4BBClaimProfessionalAndNonClinicianDiagnosisType}
+   */
+  static Coding translateLabelProfessional(DiagnosisLabel label) {
+    switch (label) {
+      case PRINCIPAL:
+        return new Coding(
+            C4BBClaimProfessionalAndNonClinicianDiagnosisType.PRINCIPAL.getSystem(),
+            C4BBClaimProfessionalAndNonClinicianDiagnosisType.PRINCIPAL.toCode(),
+            C4BBClaimProfessionalAndNonClinicianDiagnosisType.PRINCIPAL.getDisplay());
+
+        // Only Principal and Secondary are valid for this EOB type
+      case ADMITTING:
+      case FIRSTEXTERNAL:
+      case EXTERNAL:
+      case REASONFORVISIT:
+      case OTHER:
+      default:
+        return new Coding(
+            C4BBClaimProfessionalAndNonClinicianDiagnosisType.SECONDARY.getSystem(),
+            C4BBClaimProfessionalAndNonClinicianDiagnosisType.SECONDARY.toCode(),
+            C4BBClaimProfessionalAndNonClinicianDiagnosisType.SECONDARY.getDisplay());
+    }
+  }
+
+  /**
+   * Translates from {@link DiagnosisLabel} to an EOB type specific Coding based on the (@link
+   * ClaimTypeV2}.
+   */
+  static Coding translateLabelCode(DiagnosisLabel label, ClaimTypeV2 claimType) {
```

Comment:
Here is the start of the mapping from `DiagnosisLabel` to Diagnosis type on an EOB type basis.